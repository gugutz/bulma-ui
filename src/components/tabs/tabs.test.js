import React from 'react'
import renderer from 'react-test-renderer'
import Tabs from './tabs'

describe('<Tabs /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Tabs>
          <Tabs.Item isActive>Pictures</Tabs.Item>
          <Tabs.Item>Music</Tabs.Item>
          <Tabs.Item>Videos</Tabs.Item>
          <Tabs.Item>Documents</Tabs.Item>
        </Tabs>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const positions = ['right', 'left', 'center']

  for (const position of positions) {
    it('should support position prop with ' + position, () => {
      const tree = renderer
        .create(
          <Tabs position={position}>
            <Tabs.Item isActive>Pictures</Tabs.Item>
            <Tabs.Item>Music</Tabs.Item>
            <Tabs.Item>Videos</Tabs.Item>
            <Tabs.Item>Documents</Tabs.Item>
          </Tabs>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Tabs size={size}>
            <Tabs.Item isActive>Pictures</Tabs.Item>
            <Tabs.Item>Music</Tabs.Item>
            <Tabs.Item>Videos</Tabs.Item>
            <Tabs.Item>Documents</Tabs.Item>
          </Tabs>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support isToggle prop', () => {
    const tree = renderer
      .create(
        <Tabs isToggle>
          <Tabs.Item isActive>Pictures</Tabs.Item>
          <Tabs.Item>Music</Tabs.Item>
          <Tabs.Item>Videos</Tabs.Item>
          <Tabs.Item>Documents</Tabs.Item>
        </Tabs>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isToggleRounded prop', () => {
    const tree = renderer
      .create(
        <Tabs isToggle isToggleRounded>
          <Tabs.Item isActive>Pictures</Tabs.Item>
          <Tabs.Item>Music</Tabs.Item>
          <Tabs.Item>Videos</Tabs.Item>
          <Tabs.Item>Documents</Tabs.Item>
        </Tabs>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isBoxed prop', () => {
    const tree = renderer
      .create(
        <Tabs isBoxed>
          <Tabs.Item isActive>Pictures</Tabs.Item>
          <Tabs.Item>Music</Tabs.Item>
          <Tabs.Item>Videos</Tabs.Item>
          <Tabs.Item>Documents</Tabs.Item>
        </Tabs>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
