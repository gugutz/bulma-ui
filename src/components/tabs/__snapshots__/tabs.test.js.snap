// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> component should render correctly 1`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support isBoxed prop 1`] = `
<div
  className="tabs is-boxed"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support isToggle prop 1`] = `
<div
  className="tabs is-toggle"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support isToggleRounded prop 1`] = `
<div
  className="tabs is-toggle is-toggle-rounded"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support position prop with center 1`] = `
<div
  className="tabs is-centered"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support position prop with left 1`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support position prop with right 1`] = `
<div
  className="tabs is-right"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support size prop with large 1`] = `
<div
  className="tabs is-large"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support size prop with medium 1`] = `
<div
  className="tabs is-medium"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support size prop with normal 1`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Tabs /> component should support size prop with small 1`] = `
<div
  className="tabs is-small"
>
  <ul>
    <li
      className="is-active"
    >
      <a>
        Pictures
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Music
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Videos
      </a>
    </li>
    <li
      className=""
    >
      <a>
        Documents
      </a>
    </li>
  </ul>
</div>
`;
